apisix/upstream.lua

```diff
@@ -327,6 +330,7 @@ function _M.set_by_route(route, api_ctx)
 
     local nodes_count = up_conf.nodes and #up_conf.nodes or 0
     if nodes_count == 0 then
+        release_checker(up_conf.parent)
```

Comment:
Why not skip creating a new checker for upstream with no nodes?  ref: 
https://github.com/apache/apisix/blob/1d3581c95287dbab0359fb9553e1b2db5affc7bb/apisix/upstream.lua#L142

Reply:
@nic-6443 Healthchecker is not created when node == 0. This is for when previously there were non zero nodes and healthchecker was created. Later when it goes to 0, the previous healthchecker is deleted. But healthchecker is not created in case of no nodes.